src/primitives.js

```diff
@@ -212,50 +212,334 @@ export default {
+				lights: true,
+				uniforms: {
+					...UniformsLib.lights,
+					diffuse: { value: color },
+					opacity: { value: opacity },
+				},
+				vertexShader: `
+					attribute vec3 cylinderBegin;
+					attribute vec3 cylinderEnd;
+
+					varying vec3 vLightFront;
+					varying vec3 vIndirectFront;
+
+					#include <common>
+					#include <lights_pars_begin>
+
+					void main() {
+						#include <beginnormal_vertex>
+						#include <defaultnormal_vertex>
+
+						vec3 z = normalize(cylinderBegin - cylinderEnd);
+						// if z.z is 0 the cylinder doesn't appear
+						z.z += 0.0001;
+
+						vec3 x = normalize(cross(vec3(0, 1, 0), z));
+						vec3 y = cross(z, x);
+
+						float height = distance(cylinderBegin, cylinderEnd);
+
+						// position, rotate and scale the cylinder
+						mat4 cylinderMatrix = mat4(x, 0, y, 0, z * height, 0, cylinderBegin, 1);
+
+						vec4 mvPosition = modelViewMatrix * cylinderMatrix * vec4(position, 1);
+
+						gl_Position = projectionMatrix * mvPosition;
+
+						#include <lights_lambert_vertex>
+					}
+				`,
+				fragmentShader: `
+					uniform vec3 diffuse;
+					uniform float opacity;
+
+					varying vec3 vLightFront;
+					varying vec3 vIndirectFront;
+
+					#include <common>
+					#include <bsdfs>
+
+					void main() {
+						ReflectedLight reflectedLight = ReflectedLight(vec3(0), vec3(0), vec3(0), vec3(0));
+
+						reflectedLight.indirectDiffuse = vIndirectFront * BRDF_Diffuse_Lambert(diffuse);
+
+						reflectedLight.directDiffuse = vLightFront * BRDF_Diffuse_Lambert(diffuse);
+
+						gl_FragColor = vec4(reflectedLight.directDiffuse + reflectedLight.indirectDiffuse, opacity);
+					}
+				`
+			})
 		);
 
-		for (let i = 0; i < coords.length / 2; i++) {
-			const startCoordinate = new Vector3(
-				...(coords[i * 2][0] ?? scaleCoordinate(coords[i * 2][1], extent))
-			);
+		// without this the cylinders disappear when the zoom is big
+		cylinders.frustumCulled = false;
 
-			const endCoordinate = new Vector3(
-				...(coords[i * 2 + 1][0] ?? scaleCoordinate(coords[i * 2 + 1][1], extent))
-			);
+		if (edgeForm.showEdges === false) {
+			// if the edges aren't shown the work is done
+			return cylinders;
+		}
 
-			cylinders.setMatrixAt(
-				i,
-				new Matrix4()
-					.setPosition(startCoordinate)
-					.lookAt(
-						startCoordinate,
-						endCoordinate,
-						cylinders.up
-					)
-					.scale(
-						new Vector3(
-							1,
-							1,
-							// the height of the cylinder
-							startCoordinate.distanceTo(endCoordinate),
-						)
-					)
-			);
-		};
+		const group = new Group();
+
+		group.add(cylinders);
+
+		// differently from cuboid's edges, the cylinders's ones are in a different object. It is very hard or maybe impossible to draw edges with complex shapes in the fragment shader
+
+		const edgesGeometry = new InstancedBufferGeometry()
+			.setAttribute(
+				'position',
+				new BufferAttribute(
+					new Float32Array([
+						0, -1, 0,
+						0.258819043636322, -0.9659258127212524, 0,
```

Comment:
What do all of these numbers represent? How were they computed?

Reply:
I'll add a comment to it (and I'd like you to review the comment too 😁).

I got these values in the following way:
```js
console.log(new EdgesGeometry(cylinderGometry).attributes.array)
```

And then excluded the unnecessary values, without excluding them we got strange edges:
![image](https://user-images.githubusercontent.com/62714153/129801991-d915ea48-603e-4bcb-a43b-28db7bacd1ea.png)

These values are the position of each vertex of the lines.
![image](https://user-images.githubusercontent.com/62714153/129802798-000e37b1-832d-4737-bd69-c2622f349601.png)